src/datatrove/pipeline/writers/disk_base.py

```diff
@@ -40,8 +40,13 @@ def __init__(
         self.compression = compression
         self.output_folder = get_datafolder(output_folder)
         output_filename = output_filename or self.default_output_filename
-        if self.compression == "gzip" and not output_filename.endswith(".gz"):
+        if output_filename.endswith(".parquet") and compression in ["snappy", "gzip", "brotli", "lz4", "zstd"]:
+            output_filename = output_filename.replace(".parquet", f".{compression}.parquet")
+            compression = None  # compression is already handled by ParquetWriter
+        elif self.compression == "gzip" and not output_filename.endswith(".gz"):
```

Comment:
As you suggested I would rather handle this on ParquetWriter directly and pass a `compression=None` to the `super` call
```suggestion
        if self.compression == "gzip" and not output_filename.endswith(".gz"):
```

Reply:
I agree with that.